modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/executors/LCExecutor.java

```diff
@@ -0,0 +1,203 @@
+                    continue;
+                } else {
+                    setOtherRolePermission(role, currentState, targetState);
+                }
+            }
+            setPrivateRolePermission(user, currentState, targetState);
+
+            updated = true;
+
+        } catch (Exception e) {
+            String errMsg = "Failed to update Permission";
+            log.error(errMsg, e);
+            throw new Error(errMsg, e);
+        }
+
+        return updated;
+    }
+
+    /**
+     * This method is used to identify the permissions to be assigned and denied from the permission string and add or
+     * remove the permission.
+     *
+     * @param role
+     * @param permissionString
+     * @throws Exception
+     */
+    private void assignPermission(String role, String permissionString) throws Exception {
+
+        String[] permissionList = permissionString.split(",");
+        String permissionType;
+        String actionToAuthorize = "";
+        String action;
+
+        for (String permission : permissionList) {
+            if (permission.charAt(0) == '+') {
+                permissionType = ProcessCenterConstants.ALLOW;
+            } else {
+                permissionType = ProcessCenterConstants.DENY;
+            }
+            action = permission.substring(1, permission.length());
+            switch (action) {
+            case "get":
+                actionToAuthorize = ProcessCenterConstants.READ;
+                break;
+            case "add":
+                actionToAuthorize = ProcessCenterConstants.WRITE;
+                break;
+            case "delete":
+                actionToAuthorize = ProcessCenterConstants.DELETE;
+                break;
+            case "authorize":
+                actionToAuthorize = ProcessCenterConstants.AUTHORIZE;
+                break;
+            }
+            AddRolePermissionUtil.addRolePermission(userRegistry, path, role, actionToAuthorize, permissionType);
+        }
+    }
+
+    /**
+     * Assign permission for the publisher role according to the lifecycle state.
+     *
+     * @param currentState
+     * @param targetState
+     * @throws Exception
+     */
+    private void setPublisherPermission(String currentState, String targetState) throws Exception {
+
+        if (currentState.equals("Development")) {
+            assignPermission("Internal/publisher", "+get,+add,+delete,+authorize");
+
+        } else if (currentState.equals("In-Review") && targetState.equals("Development")) {
+            assignPermission("Internal/publisher", "+get,+add,+delete,+authorize");
+
+        } else if (currentState.equals("In-Review") && targetState.equals("Published")) {
+            assignPermission("Internal/publisher", "+get,+add,-delete,+authorize");
+
+        } else if (currentState.equals("Published") && targetState.equals("Development")) {
+            assignPermission("Internal/publisher", "+get,+add,+delete,+authorize");
+
```

Comment:
Fixed in https://github.com/wso2/product-pc/commit/e931bb704e9065cf57f08f61619a45a4beecb5c7